packages/salesforcedx-sobjects-faux-generator/src/describe/sObjectDescribe.ts

```diff
@@ -132,44 +133,56 @@ export class SObjectDescribe {
   }
 
   public async describeSObjectBatchRequest(
-    types: string[]
+    types: string[], cancellationToken: CancellationToken | undefined
   ): Promise<SObject[]> {
+    let batchResponse: BatchResponse;
     try {
       const batchRequest = this.buildBatchRequestBody(types);
-      const batchResponse = await this.runRequest(batchRequest);
-
-      const fetchedObjects: SObject[] = [];
-      if (batchResponse && batchResponse.results === undefined) {
-        return Promise.resolve(fetchedObjects);
-      }
-
-      batchResponse.results.forEach((sr, i) => {
-        if (sr.result instanceof Array) {
-          if (sr.result[0].errorCode && sr.result[0].message) {
-            console.log(`Error: ${sr.result[0].message} - ${types[i]}`);
-          }
-        }
-        fetchedObjects.push(sr.result);
-      });
-      return Promise.resolve(fetchedObjects);
+      batchResponse = await this.runRequest(batchRequest);
     } catch (error) {
       const errorMsg = error.hasOwnProperty('body')
         ? error.body
         : error.message;
       return Promise.reject(errorMsg);
     }
-  }
 
-  public async fetchObjects(types: string[]): Promise<SObject[]> {
-    const batchSize = MAX_BATCH_REQUEST_SIZE;
-    const requests = [];
-    for (let i = 0; i < types.length; i += batchSize) {
-      const batchTypes = types.slice(i, i + batchSize);
-      requests.push(this.describeSObjectBatchRequest(batchTypes));
+    const fetchedObjects: SObject[] = [];
+    if (
+      cancellationToken &&
+      cancellationToken.isCancellationRequested
+    ) {
+      throw new Error('SObject Refresh Cancelled');
```

Comment:
Do we need to throw an error if the request was cancelled or can we just exit early here?

Reply:
Since we are doing this as a batch process, there would be multiple requests running in parallel. So, there is a chance we end up canceling only one request and not the others. But throwing an error will force the calling method to treat all the batch requests as cancelled.

Reply:
I think it should be fine if we don't throw error and make this check after `Promise.all` call. By the time we reach here, the requests are already in progress.

That would also help to surface the actual error in the requests.

Reply:
By doing a check after Promise.all, we are waiting for all the requests to complete. By doing it here, even if some of the requests are still in progress, we would exit. I agree about we missing actual errors. Let me think about this a bit more. Thank you.

Reply:
I don't think in this scenario "we would exit" either. Consider the following code:

```
async function f() {
  try {
    const request1 = new Promise(async (resolve, reject) => {
      await new Promise((resolve) => setTimeout(resolve, 1000));
      throw new Error("interrupted");
    });
    const request2 = new Promise((resolve) => {
      setTimeout(() => {
        console.log("Im the long running request");
        resolve();
      }, 5000);
    });
    await Promise.all([request1, request2]);
  } catch (error) {}
}

f();
```
You should still be able to see "Im the long running request" output even after it has thrown an error. Once the request has been made, it generally wouldn't be cancellable.

Reply:
Thank you @xyc!
That makes sense, so I could just add a check after Promise.all.

Reply:
`async function f() {
  try {
    const request1 = new Promise(async (resolve, reject) => {
      await new Promise((resolve) => setTimeout(resolve, 1000));
      throw new Error("interrupted");
    });
    const request2 = new Promise((resolve) => {
      setTimeout(() => {
        console.log("Im the long running request");
        resolve();
      }, 5000);
    });
    await Promise.all([request1, request2]);
  } catch (error) {
      console.log("I have exited!");
  }
}

f();`

Just one more thing - What I was trying is not to cancel a request that has been made. If the "I have exited" is printed before "Im the long running request", it means we are cancelling without waiting for all requests to complete. But again this is not worth the complexity it brings. So, will just check after Promise.all like you suggested.

Reply:
that's right, but we should note that once cancel button is clicked the dialog is dismissed immediately and we are handling this in background here.